dotnet/src/Connectors/Connectors.Memory.SqliteVec/SqliteFilterTranslator.cs

```diff
@@ -61,20 +61,11 @@ protected override void TranslateQueryParameter(string name, object? value)
         }
         else
         {
-            // Duplicate parameter name, create a new parameter with a different name
-            // TODO: Share the same parameter when it references the same captured value
-            if (this._parameters.ContainsKey(name))
-            {
-                var baseName = name;
-                var i = 0;
-                do
-                {
-                    name = baseName + (i++);
-                } while (this._parameters.ContainsKey(name));
-            }
-
-            this._parameters.Add(name, value);
-            this._sql.Append('@').Append(name);
+            // The param name is just the index, so there is no need for escaping or quoting.
```

Comment:
It mimics what the two other SQL translator do. I know it's not perfect, but it's simple and it always works and we have little of time.